python/taichi/main.py

```diff
@@ -255,97 +261,318 @@ def make_argument_parser():
+        x_end = float(sys.argv[4])
+        y_begin = float(sys.argv[5])
+        y_end = float(sys.argv[6])
+        crop_video(input_fn, output_fn, x_begin, x_end, y_begin, y_end)
+
+    def video_scale(self, arguments: list = sys.argv[2:]):
+        """Scale video resolution"""
+        # TODO: Convert the logic to use args
+        parser = argparse.ArgumentParser(description=f"{self.video_scale.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
+
+        input_fn = sys.argv[2]
+        assert input_fn[-4:] == '.mp4'
+        output_fn = input_fn[:-4] + '-scaled.mp4'
+        ratiow = float(sys.argv[3])
+        if len(sys.argv) >= 5:
+            ratioh = float(sys.argv[4])
         else:
-            ret = test_python(args)
-            if ret != 0:
-                return ret
-            return test_cpp(args)
-    elif mode == "benchmark":
+            ratioh = ratiow
+        scale_video(input_fn, output_fn, ratiow, ratioh)
+
+    def video(self, arguments: list = sys.argv[2:]):
+        """Make a video using *.png files in the current folder"""
+        # TODO: Convert the logic to use args
+        parser = argparse.ArgumentParser(description=f"{self.video.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
+
+        files = sorted(os.listdir('.'))
+        files = list(filter(lambda x: x.endswith('.png'), files))
+        if len(sys.argv) >= 3:
+            frame_rate = int(sys.argv[2])
+        else:
+            frame_rate = 24
+        if len(sys.argv) >= 4:
+            trunc = int(sys.argv[3])
+            files = files[:trunc]
+        ti.info('Making video using {} png files...', len(files))
+        ti.info("frame_rate={}", frame_rate)
+        output_fn = 'video.mp4'
+        make_video(files, output_path=output_fn, frame_rate=frame_rate)
+        ti.info('Done! Output video file = {}', output_fn)
+
+    def doc(self, arguments: list = sys.argv[2:]):
+        """Build documentation"""
+        # TODO: Convert the logic to use args
+        parser = argparse.ArgumentParser(description=f"{self.doc.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
+
+        os.system('cd {}/docs && sphinx-build -b html . build'.format(
+            ti.get_repo_directory()))
+
+    def interpolate(self, arguments: list = sys.argv[2:]):
+        """???"""
+        # TODO: Convert the logic to use args
+        parser = argparse.ArgumentParser(description=f"{self.interpolate.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
+        interpolate_frames('.')
+
+    def asm(self, arguments: list = sys.argv[2:]):
+        """???"""
+        # TODO: Convert the logic to use args
+        parser = argparse.ArgumentParser(description=f"{self.asm.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
+
+        fn = sys.argv[2]
+        os.system(
+            r"sed '/^\s*\.\(L[A-Z]\|[a-z]\)/ d' {0} > clean_{0}".format(fn))
+
+    def update(self, arguments: list = sys.argv[2:]):
+        """???"""
```

Comment:
Sorry about the confusion. This is for updating the Taichi codebase. Not sure if it still works though...

Reply:
No worries! Let me test it, if it stops working, will mark it in the helper msg and fix it (probably in a separate PR)